SU2_CFD/src/solvers/CHeatSolver.cpp

```diff
@@ -392,10 +392,10 @@ void CHeatSolver::Upwind_Residual(CGeometry *geometry, CSolver **solver_containe
           Vector_j[iDim] = 0.5*(geometry->nodes->GetCoord(iPoint, iDim) - geometry->nodes->GetCoord(jPoint, iDim));
         }
 
-        Gradient_i = solver_container[FLOW_SOL]->GetNodes()->GetGradient_Reconstruction(iPoint);
-        Gradient_j = solver_container[FLOW_SOL]->GetNodes()->GetGradient_Reconstruction(jPoint);
-        Temp_i_Grad = nodes->GetGradient_Reconstruction(iPoint);
-        Temp_j_Grad = nodes->GetGradient_Reconstruction(jPoint);
+        const auto Gradient_i = solver_container[FLOW_SOL]->GetNodes()->GetGradient_Reconstruction(iPoint);
+        const auto Gradient_j = solver_container[FLOW_SOL]->GetNodes()->GetGradient_Reconstruction(jPoint);
+        const auto Temp_i_Grad = nodes->GetGradient_Reconstruction(iPoint);
+        const auto Temp_j_Grad = nodes->GetGradient_Reconstruction(jPoint);
```

Comment:
Temp_i_Grad already defined in this scope (on the "block above") a few lines above. maybe `Temp_i_Grad_Recon` ? Isnt this somewhat dangerous because if someone deletes these lines (for some reason) there will still be a a valid Temp_i_Grad object floating around that does not return what one actually wants? ... one of  the worst kinds of error

Reply:
I may have missed it. I'll circle back to this.

Reply:
Well, before the scope of that variable was even larger... This type of reconstruction logic should be shared by scalar solvers in the future, so I won't lose too much time with this.